src/toast/tests/ops_polyfilter.py

```diff
@@ -189,22 +189,22 @@ def test_polyfilter2D(self):
 
         # Plot unfiltered TOD
 
-        if data.comm.world_rank == 0:
-            set_matplotlib_backend()
-            import matplotlib.pyplot as plt
-
-            fig = plt.figure(figsize=[18, 12])
-            ax = fig.add_subplot(1, 2, 1)
-            ob = data.obs[0]
-            for idet, det in enumerate(ob.local_detectors):
-                flags = np.array(ob.shared["flags"])
-                flags |= ob.detdata["flags"][det]
-                good = flags == 0
-                signal = ob.detdata["signal"][det]
-                x = np.arange(signal.size)
-                ax.plot(x, signal, "-", label=f"{det} unfiltered")
-                ax.plot(x, good, "-", label=f"{det} input good samples")
-            ax.legend(loc="best")
+        # if data.comm.world_rank == 0:
```

Comment:
I don't think the issue is these matplotlib commands. I was able to hang the 2D polyfilter test on my laptop without the plotting commands. The precise location of the hang also seems to depend on the order of the polyfilter unit tests. Could be some sort of memory addressing error.